pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UseUnderscoresInNumericLiterals.xml

```diff
@@ -0,0 +1,280 @@
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    private callBar() {
+        bar(314_159_265L);
+    }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, String value with '_'
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    String pi = "zcx_xvc_cbv_vnb_bmn";
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, String value without '_'
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    String pi = "3639";
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal in binary
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 0b100111;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal in octal
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 047474747;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal in hexadecimal
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 0xabcdef;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal in negative binary
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = -0b100111;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal in negative octal
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = -047474747;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal in negative hexadecimal
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = -0xabcdef;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal with exponent
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 39e1234;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
+ok, Numeric Literal with negative exponent
+     ]]></description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    int binary = 36e-11234;
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description><![CDATA[
```

Comment:
Please don't wrap the description of tests in a CDATA block. They nearly never require it and it makes the test file harder to read.